testutil/testdevice_linux.go

```diff
@@ -0,0 +1,97 @@
+// +build linux
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type testDevice struct {
+	opts       TestDeviceOpts
+	fileName   string
+	deviceName string
+}
+
+func NewTestDevice(t *testing.T, opts TestDeviceOpts) TestDevice {
```

Comment:
Ah, okay, I see.

So, we all this `NewLoopback`. Leave the mkfs out of this: the caller should just get a new block device that they can do whatever with. Does that make sense?